easypoll/src/main/java/com/eis0/easypoll/poll/PollManager.java

```diff
@@ -0,0 +1,174 @@
+                .addArgument(poll.getName())
+                .addArgument(poll.getQuestion())
+                .addArguments(usersNetwork.getAddresses());
+        usersNetwork.broadcastMessage(builder.buildMessage());
+        dataProvider.addPoll(poll);
+    }
+
+    // ---------------------------- POLL ANSWERING ---------------------------- //
+
+    /**
+     * Send the answer to the author and all the users subscribed to the poll. Finally, remove
+     * the answered poll from the incoming polls list.
+     *
+     * @param poll The poll to answer.
+     * @param answer The user's answer, true equals "Yes" and false equals "No".
+     * @throws IllegalArgumentException If the author is trying to answer an owning poll.
+     * @author Matteo Carnelos
+     */
+    public void answerPoll(@NonNull BinaryPoll poll, boolean answer) {
+        if(poll.getAuthor().isLocalNetwork())
+            throw new IllegalArgumentException("Authors are not allowed to answer their polls.");
+        String number = String.valueOf(poll.getNumber());
+        String answerCode = answer ? PollCommands.YES_ANSWER : PollCommands.NO_ANSWER;
+        // The ANSWER_POLL message for users follows this pattern:
+        // ANSWER_POLL_CMD + number + authorAddress + answerCode
+        NetworkMessageBuilder builder = new NetworkMessageBuilder(PollCommands.ANSWER_POLL)
+                .addArgument(number)
+                .addArguments(poll.getAuthor().getAddresses())
+                .addArgument(answerCode);
+        poll.getUsers().broadcastMessage(builder.buildMessage());
+        // The ANSWER_POLL message for author follows this pattern:
+        // ANSWER_POLL_CMD + number + LOCALNET_ADDR + answerCode
+        // The LOCALNET_ADDR tells the author that the answer is for a poll that he created
+        builder = new NetworkMessageBuilder(PollCommands.ANSWER_POLL)
+                .addArgument(number)
+                .addArgument(Network.LOCALNET_ADDR)
+                .addArgument(answerCode);
+        poll.getAuthor().broadcastMessage(builder.buildMessage());
+        DataProvider.getIncomingPolls().remove(poll);
+    }
+
+    // ---------------------------- POLL RECEIVING ---------------------------- //
+
+    /**
+     * Receives an {@link SMSMessage} and updates polls data accordingly.<br>
+     * Every message contains a command line, composed by a command and a variable number of arguments.
+     * The command is always one of the following values:<br>
+     * NEW_POLL_CMD - When the message contains a new poll;<br>
+     * ANSWER_POLL_CMD -  When the message contains an answer.<br>
+     * The number and meaning of the arguments depends on the type of command.
+     *
+     * @param message The SMS message passed by {@link SMSManager}. The message is formatted following
+     *                the rules of the {@link NetworkMessageBuilder}.
+     * @author Matteo Carnelos
+     */
+    public void onMessageReceived(SMSMessage message) {
```

Comment:
I was thinking that maybe it would have been better to decrease the length of this method using two auxiliary private methods that substituted the content of the two cases in the switch structure, remembering that a code is more fluent in its reading when methods have short length.

Reply:
You're right, this change would increase code readability, I will surely do that